File: DigiMox/app/src/main/java/com/digimox/adapters/DMMainCategoryPagerAdapter.java
package com.digimox.adapters;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Deepesh on 10-Dec-15.
 */
public class DMMainCategoryPagerAdapter extends FragmentStatePagerAdapter {
    private List<Fragment> mFragmentList;

    public DMMainCategoryPagerAdapter(FragmentManager fm) {
        super(fm);
        mFragmentList = new ArrayList<Fragment>();
    }

    public void addFragment(int position, Fragment fragment) {
        mFragmentList.add(position, fragment);
    }

    public Fragment getFragment(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public int getCount() {
        return mFragmentList.size();
    }

    @Override
    public Fragment getItem(int position) {
        return mFragmentList.get(position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return "";
    }
}

File: DigiMox/app/src/main/java/com/digimox/adapters/DMMainCategoryPagerAdapter.kt
package com.digimox.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

import java.util.ArrayList

/**
 * Created by Deepesh on 10-Dec-15.
 */
class DMMainCategoryPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm) {
    private val mFragmentList: MutableList<Fragment>

    init {
        mFragmentList = ArrayList()
    }

    fun addFragment(position: Int, fragment: Fragment) {
        mFragmentList.add(position, fragment)
    }

    fun getFragment(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getCount(): Int {
        return mFragmentList.size
    }

    override fun getItem(position: Int): Fragment {
        return mFragmentList[position]
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return ""
    }
}
